BlockSettleUILib/BSTerminalMainWindow.cpp

```diff
@@ -2604,13 +2620,33 @@ void BSTerminalMainWindow::autoSignDeliveryObligation(const ProxyTerminalPb::Res
             logMgr_->logger()->debug("[BSTerminalMainWindow::autoSignDeliveryObligation] ZC pushed, id={}", pushId);
          }
       };
-      signContainer_->signTXRequest(txReq, cbSigned, SignContainer::TXSignMode::AutoSign);
+      signContainer_->signTXRequest(txReq, cbSigned
+         , (autoSignQuoteProvider_->autoSignState() == bs::error::ErrorCode::NoError)
+         ? SignContainer::TXSignMode::AutoSign : SignContainer::TXSignMode::Full);
    };
    utxoReservationMgr_->reserveBestXbtUtxoSet(walletsMgr_->getPrimaryWallet()->walletId()
       , std::abs(request.to_deliver()) + (uint64_t)(utxoReservationMgr_->feeRatePb() * 321)
       , false, cbInputs, false, bs::UTXOReservationManager::CheckAmount::Enabled, true);
 }
 
+bool BSTerminalMainWindow::canSendObligationTX() const
+{
+   if (autoSignQuoteProvider_->autoSignState() == bs::error::ErrorCode::NoError) {
+      return true;
+   }
+   const auto& priWallet = walletsMgr_->getPrimaryWallet();
+   if (!priWallet) {
+      logMgr_->logger()->debug("[{}] no primary wallet", __func__);
+      return false;
+   }
+   const auto& encTypes = priWallet->encryptionTypes();
+   if (/*(priWallet->encryptionRank().n == 1) &&*/ !encTypes.empty()
```

Comment:
Please remove commented code.

Reply:
I better uncommented it to indicate this is applicable only to n=1 rank